ComplianceAsCode/auditree-arboretum
# new feature: create OSCAL json report from compliance operator evidence

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)


## What
Transform Kubernetes compliance operator evidence from cluster_resource fetcher into a NIST OSCAL Assessment Results collection of Observations in JSON format.

## Why

Rationale: standardized version of evidence for multi-cloud and to facilitate creation of NIST OSCAL Assessment Results.

## How

Write a harvest report `compliance_oscal_observations` that consumes `cluster_resource.json` evidence and optional `oscal-metadata.yaml` to produce `compliance_oscal_observations.json`.

## Test

- Ensure check results summary filename property is set.
- Ensure proper handling of no cluster_resource content.
- Ensure proper report creation.
- Ensure proper report creation with oscal metadata.
- Test no 'resources' found.
- Test no 'kind' found.
- Test no 'ConfigMap' found.
- Test no 'data' found.
- Test no 'results' found.
- Test no 'metadata' found.
- Test no 'name' found.

## Context

- [49](https://github.com/ComplianceAsCode/auditree-arboretum/issues/49)
